quiz.go

```diff
@@ -21,31 +27,41 @@ func main() {
 
 	r := csv.NewReader(bufio.NewReader(file))
 
-	for {
-		record, err := r.Read()
-		if err == io.EOF {
-			break
-		}
-		if err != nil {
-			log.Fatal(err)
-			continue
-		}
+	go func(r *csv.Reader, correct, incorrect *int) {
+		for {
+			record, err := r.Read()
+			if err == io.EOF {
+				finished <- true
+			}
+			if err != nil {
+				log.Fatal(err)
+			}
 
-		problem := record[0]
-		solution := record[1]
+			problem := record[0]
+			solution := record[1]
 
-		fmt.Printf("%v?\n", problem)
-		var response string
-		fmt.Scanln(&response)
+			fmt.Printf("%v?\n", problem)
+			var response string
+			fmt.Scanln(&response)
 
-		if response == solution {
-			fmt.Println("Correct!")
-			numCorrect++
-		} else {
-			fmt.Println("Incorrect :(")
-			numIncorrect++
+			if response == solution {
+				fmt.Println("Correct!")
+				*correct++
+			} else {
+				fmt.Println("Incorrect :(")
+				*incorrect++
+			}
 		}
-	}
+	}(r, &numCorrect, &numIncorrect)
+
+	timeLimit := time.After(30 * time.Second)
 
-	fmt.Printf(" Number correct: %v, number incorrect: %v\n", numCorrect, numIncorrect)
+	select {
+	case <-timeLimit:
+		fmt.Println("TIMES UP YER DONE")
+		printResults(numCorrect, numIncorrect)
```

Comment:
does this have to be inside the `select` statement?

Reply:
no good call